### DISCUSSION TITLE
Allow branch deletions to certains users

### DISCUSSION THREAD

**Discussion by RegFacu** [community/community#38179] [Repositories]
Currently a protected branch can't be deleted by default as it says on your [documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#allow-deletions). On a release process perspective, I think that is great for regular users.

The thing is that an user as a release manager who is not an admin but could bypass some protection rules should be able to also delete a protected branch if he considers something was wrong and he needs to create the branch again.

Could something like `Allow force pushes` option be added?

<img width="753" alt="image" src="https://user-images.githubusercontent.com/13556129/199754102-bda6ded3-2f1e-449f-877b-9425a3b46232.png">

**Comment by david-engelmann** [comment]
[Delete Branch protections are described in the detail in this part of the documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#deleting-a-branch-protection-rule)

**Comment by RegFacu** [comment]
Right, but I don't want to delete the branch protection. I want to be able to delete a branch while a branch protection is active.

**Reply by david-engelmann** [reply]
The link is to info on protections around deleting branches

**Reply by RegFacu** [reply]
Yes, I know, but I'm asking to improve those protections

**Comment by RegFacu** [comment]
Any comment?

**Comment by bturner-cpacket** [comment]
I would love to see improvements here...

We have branch protections that look something like this:

`feature/*`
`release/*`

If I have `release/1`, `release/2`, and `release/3` branches and it turns out that release/3 was created prematurely or in error, I have to do the following to fix that:

1. Note down all the settings for the branch protection `release/*`
2. Delete the branch protection
3. Delete `release/3` branch
4. re-create the branch protection for `release/*` with the same settings it had before

What I think would make sense is the ability to delete a branch that has branch protections by being able to bypass the branch protections if you are an administrator.  So for example administrators can check the box `Merge without waiting for requirements to be met (bypass branch protections)` to merge a pull request, I would love to see a similar box that allows something like `Bypass branch protections and delete branch release/3`

Thanks
